x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/new_terms/bulk_create_suppressed_alerts_in_memory.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { SuppressedAlertService } from '@kbn/rule-registry-plugin/server';
+
+import type { SuppressionFieldsLatest } from '@kbn/rule-registry-plugin/common/schemas';
+import type {
+  SearchAfterAndBulkCreateParams,
+  SearchAfterAndBulkCreateReturnType,
+  WrapSuppressedHits,
+} from '../types';
+
+import type { AlertSuppressionCamel } from '../../../../../common/api/detection_engine/model/rule_schema';
+import { DEFAULT_SUPPRESSION_MISSING_FIELDS_STRATEGY } from '../../../../../common/detection_engine/constants';
+import { AlertSuppressionMissingFieldsStrategyEnum } from '../../../../../common/api/detection_engine/model/rule_schema';
+
+import { executeBulkCreateAlerts } from '../utils/bulk_create_suppressed_alerts_in_memory';
+import type {
+  BaseFieldsLatest,
+  WrappedFieldsLatest,
+  NewTermsFieldsLatest,
+} from '../../../../../common/api/detection_engine/model/alerts';
+import { partitionMissingFieldsEvents } from '../utils/partition_missing_fields_events';
+import type { EventsAndTerms } from './types';
+import type { ExperimentalFeatures } from '../../../../../common';
+
+interface SearchAfterAndBulkCreateSuppressedAlertsParams extends SearchAfterAndBulkCreateParams {
+  wrapSuppressedHits: WrapSuppressedHits;
+  alertTimestampOverride: Date | undefined;
+  alertWithSuppression: SuppressedAlertService;
+  alertSuppression?: AlertSuppressionCamel;
+}
+export interface BulkCreateSuppressedAlertsParams
+  extends Pick<
+    SearchAfterAndBulkCreateSuppressedAlertsParams,
+    | 'bulkCreate'
+    | 'services'
+    | 'ruleExecutionLogger'
+    | 'tuple'
+    | 'alertSuppression'
+    | 'alertWithSuppression'
+    | 'alertTimestampOverride'
+  > {
+  wrapHits: (
+    events: EventsAndTerms[]
+  ) => Array<WrappedFieldsLatest<BaseFieldsLatest & NewTermsFieldsLatest>>;
+  wrapSuppressedHits: (
+    events: EventsAndTerms[]
+  ) => Array<
+    WrappedFieldsLatest<BaseFieldsLatest & SuppressionFieldsLatest & NewTermsFieldsLatest>
+  >;
+  eventsAndTerms: EventsAndTerms[];
+  toReturn: SearchAfterAndBulkCreateReturnType;
+  experimentalFeatures: ExperimentalFeatures | undefined;
+}
+/**
+ * bulk create and suppress alerts in memory,
```

Comment:
Nit: these comments don't provide much more information than the name does. I would mention that this is explicitly for new terms alerts, or at least deals with `EventsAndTerms[]` instead of normal alerts.

The note about missing fields logic is nice, but it would be nice to discuss it more explicitly, i.e. "if {@param alertSuppression.missingFieldsStrategy} is X, alerts missing the suppression field(s) will not be suppressed."

But: typing out the above, this doesn't feel like the right place to document that behavior. I would capture it in documentation, and ideally a unit test for this function, otherwise an integration test of some kind.

Reply:
I added a bit more details to description, mentioned difference with common utility.

There multiple FTR tests that cover missing fields logic: https://github.com/elastic/kibana/blob/ff8a41fdf43e748c214b6b1e535aaac4bc28669a/x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/rule_execution_logic/trial_license_complete_tier/execution_logic/new_terms_alert_suppression.ts#L874-L981